File: solutions/erlang/darts/1/src/darts.erl
-module(darts).

-export([score/2]).


score(X, Y) -> score_distance_squared(X * X + Y * Y).

score_distance_squared(RR) when RR > 10 * 10 -> 0;
score_distance_squared(RR) when RR > 5 * 5 -> 1;
score_distance_squared(RR) when RR > 1 -> 5;
score_distance_squared(_RR) -> 10.

File: solutions/go/darts/1/darts.go
package darts
import "math"

func Score(x, y float64) int {
    dist := math.Sqrt(x * x + y * y)
    if dist <= 1.0 { return 10 }
    if dist <= 5.0 { return 5 }
    if dist <= 10.0 { return 1 }
	return 0
}
